x/group/msgs.go

```diff
@@ -231,20 +225,16 @@ func NewMsgCreateGroupWithPolicy(admin string, members []Member, group_metadata
 	return m, nil
 }
 
-func (m *MsgCreateGroupWithPolicy) GetDecisionPolicy() DecisionPolicy {
+func (m *MsgCreateGroupWithPolicy) GetDecisionPolicy() (DecisionPolicy, error) {
 	decisionPolicy, ok := m.DecisionPolicy.GetCachedValue().(DecisionPolicy)
 	if !ok {
-		return nil
+		return nil, sdkerrors.ErrInvalidType.Wrapf("expected %T, got %T", (DecisionPolicy)(nil), m.DecisionPolicy.GetCachedValue())
```

Comment:
I thought that returning a `nil` when the interface was not satisfied was error-prone. Proposing to return an error